### ISSUE TITLE
[Bug]: Configuration for locations loads twice 

### ISSUE THREAD

**Issue by vaughanknight** [Green-Software-Foundation/carbon-aware-sdk#535]
### What happened?

The configuration for locations loads twice at startup, it's visible in the logs with suffixed locations being loaded at startup i.e. westus, and then westus_1.  It happens for all configured location files.

### client

WebAPI (Default)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by YaSuenag** [comment]
I saw same issue, and I created a patch (but I've not sent it as a PR): https://github.com/YaSuenag/carbon-aware-sdk/commit/84662a41513a0f160b4a210ff676f276fbce5bf3

Maybe it happens when both EmissionsDataSource and ForecastDataSource are defined.  
LocationSource would be loaded when each data source are prepared, so same location JSON would be loaded by preparation phase of both EmissionsDatasource and ForecastDataSource.

My patch adds `HashSet` to remember loaded location JSONs, so it prevents to load in twice.

I can send PR if you want.
